### DISCUSSION TITLE
Use of image loader (with env var & custom server)

### DISCUSSION THREAD

**Discussion by orbiteleven** [vercel/next.js#22869] [Help] [answered]
Hello,

I'm a bit stumped here. I'm trying to use [ImgIx](https://www.imgix.com) as my image loader. Things work fine on my development server, and when hard-coding the `images.path`, but if I try to use an environment variable it fails. I can see in the logs (Heroku) that the env var exists at load time, so I'm wondering if it has to do with running a (fairly standard) custom server:

```javascript
import dotenv from 'dotenv'
dotenv.config()

import express from 'express'
import next from 'next'

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev })
const handle = app.getRequestHandler()
const host = process.env.HOST || 'localhost'
const port = process.env.PORT || 3000

;(async () => {
  try {
    await app.prepare()
    const server = express()
    server.get('/health', (req, res) => res.send('ok'))
    server.all('/api/graphql', handle)

    server.all('*', handle)
    server.listen(port, host, (err) => {
      if (err) throw err
      console.log(`> Ready on ${host}:${port} - env ${process.env.NODE_ENV}`)
    })
  } catch (e) {
    console.error(e)
    process.exit(1)
  }
})()
```

I have the following `next.config.js` setup:

```javascript
/* eslint-disable @typescript-eslint/no-var-requires*/
const withPlugins = require('next-compose-plugins')
const withGraphql = require('next-graphql-loader')

const { PHASE_PRODUCTION_BUILD } = require('next/constants')

console.log('***In next.config.js ', process.env.IMAGE_LOADER_URL)

module.exports = withPlugins([withGraphql], {
  [PHASE_PRODUCTION_BUILD]: {
    webpack: (config, { webpack }) => {
      config.plugins.push(new webpack.IgnorePlugin(/\/__tests__\//))
      return config
    },
  },
  images: {
    loader: 'imgix',
    path: process.env.IMAGE_LOADER_URL,
  },
})
```

**Comment by orbiteleven** [comment, marked as answer]
So this was twofold:

 1. Next.js reads this file BEFORE loading any `.env` files on build.
 2. Heroku does not run `docker build` with env vars while running containerized builds.

Ultimately my solution was to abandon Dockerized builds. This also increased my deploy speed.

**Reply by vicentereig** [reply]
@orbiteleven Kind of a tangential question here, as we ran in a similar scenario: did you set up your Source as web proxy or a web folder?

We followed the instructions in the Nextjs docs to setup the imgix integration, and seems that the `s=` with the token is missing.  Are you signing URLs on the serve-side/at build time with their client? https://docs.imgix.com/libraries/js-core

**Reply by orbiteleven** [reply]
Hey @vicentereig. Neither. We're using the S3 bridge and not using signed URLs (yet). 

TBH we might move away from `next/image` so we have better control over our image generation, at which point we'd implement singed URLs... but that's equally tangential :)
